### ISSUE TITLE
Surround sound upmix - feature request

### ISSUE THREAD

**Issue by ujLion** [duncanthrax/scream#58]
I am currently using a 5.1 to 5.1 configuration.
The destination computer supports "speaker fill" - however it needs a 2.1 input signal to engage.
But since the input signal for it is 5.1 - the 'upmixing' does not engage.

The work around is to fallback to a 2.1 to 5.1 configuration. Then upon restarting the audio stream - the 'upmixing' does engage as expected. But this requires switching between 2.1 and 5.1 from source depending on the media being played - which is very inconvenient.

It would be nice to have the upmix available in the virtual sound card driver itself.
I understand that this may be beyond the scope of what this project seeks to do.

Reference material: https://chunminchang.github.io/blog/post/audio-5-1

**Comment by duncanthrax** [comment]
This is hard to fit into the way scream currently works (dumping Windows sound subsystem output straight to network or shared memory).

Also my instinct tells me that this is very much a client issue :)

**Comment by ujLion** [comment]
Windows provides: "Audio Effects" interface for the audio driver. Looks like it will handle the required task on its own. We only need to engage it. https://docs.microsoft.com/en-us/windows-hardware/drivers/audio/audio-signal-processing-modes

**Comment by ujLion** [comment]
True that - processing wise, this could be done on the client side. By detecting that the other channels are empty. However if this feature is provided by the OS to the driver itself - I think using it directly would be the way to go, should be easier.

**Comment by duncanthrax** [comment]
You are right, it has "Speaker Fill" ... hmmm. I'll check if it is really that easy to register the driver for these APOs.

**Comment by Gandalf1783** [comment]
I just wanted to mention that you could use Voicemeeter for Windows. There you can upmix to 5.1, 7.1 and do "Channel Repeat". You have 2 to 8 output devices (depending on chosen Version).
I do not want to advert it here, but i thought it would fit here.
